### PR TITLE
Distribute random source uniformly by solid angle

### LINKED ISSUE #328: Draw random point sources uniformly distributed over solid angle.
https://github.com/RadioAstronomySoftwareGroup/pyuvsim/blob/d571b4847b1bf884236f13994a2d79f429c8483a/pyuvsim/simsetup.py#L249

random sources are currently drawn distributed uniformly in azimuth and altitude but to get a uniform distribution per solid angle, one should actually do something like this

https://mathworld.wolfram.com/SpherePointPicking.html

### PR DESCRIPTION
**Pull request by aelanman**
## Description

Modifies the random distribution mock catalog to ensure sources are distributed uniformly by solid angle.

## Motivation and Context

The existing implementation of random sources chooses uniformly-distributed altitudes and azimuths, which leads to a nonuniform distribution of points on the sphere (clustering near the poles). To ensure the random points are truly distributed uniformly on the sphere, the distribution of altitude had to be adjusted.

Closes #328 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Reference simulation update or replacement
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

For all pull requests:
- [x] I have read the [contribution guide](CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Bug fix checklist:
- [x] My fix includes a new test that breaks as a result of the bug (if possible).
- [x] All new and existing tests pass.
- [ ] I have checked that I reproduce the reference simulations or if there are differences they are explained below (if appropriate). If there are changes that are correct, I will update the reference simulation files after this PR is merged.
- [ ] I have checked (e.g., using the benchmarking tools) that this change does not significantly increase typical runtimes. If it does, I have included a justification in the comments on this PR.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvsim/blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by bhazelton** [comment]
Was this random distribution used in any of our reference sims?

**Comment by aelanman** [comment]
It's not in any ref sims.

File: pyuvsim/tests/test_simsetup.py [hunk 1]
```diff
@@ -884,48 +884,91 @@ def test_uvfits_to_config():
...
+    [
+        ('cross', 'mock_cross_2458098.27471.txt'),
+        ('hera_text', 'mock_hera_text_2458098.27471.txt'),
+        ('long-line', 'mock_long-line_2458098.27471.txt'),
+        ('off-zenith', 'mock_off-zenith_2458098.27471.txt'),
+        ('triangle', 'mock_triangle_2458098.27471.txt'),
+        ('random', 'mock_random_2458098.27471.txt'),
+        ('zenith', 'mock_zenith_2458098.27471.txt'),
+    ]
+)
```
Comment (bhazelton) [inline pyuvsim/tests/test_simsetup.py:898]: nice use of parametrize!

File: pyuvsim/tests/test_simsetup.py [hunk 2]
```diff
@@ -884,48 +884,91 @@ def test_uvfits_to_config():
...
+    hp = astropy_healpix.HEALPix(nside)
+
+    inds = hp.lonlat_to_healpix(Longitude(az, unit='rad'), Latitude(alt, unit='rad'))
+    un, counts = np.unique(inds, return_counts=True)
+
+    # counts should be Poisson-distributed with rate lambda = nsrcs / npix
+    # variance and mean should be close to lambda
+    lam = Nsrcs / npix
+    assert np.isclose(np.mean(counts), lam, atol=1.0)
+    assert np.isclose(np.var(counts), lam, atol=1.0)
```
Comment (bhazelton) [inline pyuvsim/tests/test_simsetup.py:965]: This is a clever way to handle testing a random distribution. The only thing I worry about slightly with random things in tests are flaky tests that fail randomly. Do you have a sense for how likely this is to fail with these tolerances?

**Review by bhazelton — APPROVED** [review]
Looks pretty good to me, as long as the test for the random distribution is unlikely to fail in a flaky way.

File: pyuvsim/tests/test_simsetup.py (hunk 2 above)
Comment (aelanman) [reply pyuvsim/tests/test_simsetup.py:965]: The random generator is given the same seed for each run, so it should always turn up the same points. That said, I just ran it 500 times without the fixed rseed, so it's actually random, and didn't have any failures.

File: pyuvsim/tests/test_simsetup.py (hunk 1 above)
Comment (aelanman) [reply pyuvsim/tests/test_simsetup.py:898]: Thanks :)

File: pyuvsim/tests/test_simsetup.py (hunk 2 above)
Comment (bhazelton) [reply pyuvsim/tests/test_simsetup.py:965]: Ah, I didn't catch that it was seeded. I'm good with that, thanks.

**Pull request merged**